### PR TITLE
ENG-341 - Add IconFrame and FormTitle components

### PR DESCRIPTION
**Pull request by floreks**
- Added `Icon` component with more variants
  - Sizes: `xsmall | small | medium | large | xlarge`
  - Spacing: `none | padding`
  - Hue: `default | lighter | lightest`
  - Updated stories to cover more scenarios
- Added `critical` severity to `Chip` component
- Added `FormTitle` component

### CODE REVIEW

**Comment by dogmar** [comment]
`Icon` seems like too general a name for this, since I can imagine we may need different "icon"-ish components in the future. Also the word "Icon" is currently only used for graphic components in the "/icons" section of the library, and I think we should stick to only using it there to avoid confusion. The figma file calls it an 'App Icon' so what about `AppThumbnail`?

File: src/components/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  const color = hueToColor[hue]
+  const borderColor = hueToBorderColor[hue]
+
+  return (
+    <Flex
+      backgroundColor={color}
+      borderRadius="medium"
+      border="1px solid border"
+      borderColor={borderColor}
+      width={boxSize}
```
Comment (dogmar) [inline src/components/Icon.tsx:72]: Need to add an `overflow: hidden` so border-radius will properly crop 100% width/height images. Currently you're relying on the built-in radius in the images, but we should make sure it works for all images. Also, in the figma file, full-width images do not have the 1px border, so you'll need to account for that variant in some way.

**Review by dogmar — CHANGES_REQUESTED** [review]

File: src/components/Icon.tsx (hunk 1 above)
Comment (floreks) [reply src/components/Icon.tsx]: Good catch. Fixed.

**Comment by floreks** [comment]
Renamed the component to `IconFrame` as in figma and fixed found issues. PTAL

File: src/stories/Icon.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
+import { Flex, H3 } from 'honorable'
```
Comment (dherault) [inline src/stories/Icon.stories.tsx:1]: Should be renamed IconFrame.stories.jsx

File: src/stories/Icon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+import IconFrame from '../components/IconFrame'
+
+export default {
+  title: 'Icon',
```
Comment (dogmar) [inline src/stories/Icon.stories.tsx]: Change this to 'Icon Frame' and I'm good. :)

Comment (floreks) [reply src/stories/Icon.stories.tsx]: Oh man... This is not my day :slightly_smiling_face:

**Review by dogmar — APPROVED** [review]

**Pull request merged**